### DISCUSSION TITLE
trail whitespaces, style convention and how to act

### DISCUSSION THREAD

**Discussion by mglmc** [edgexfoundry/community#139] [Q&A] [answered]
Hi, 

I found some issues that I'm going to report or do a PR for the project.

While I was working on that, I found that the file is full of trail whitespace and other minor things that don't match with a consistent style.

I usually work that if I change a file, the styling has to be consistent in the whole file.

But, I don't know the rules of this project for those cases.

Possibilities:

- Create a PR just for style inconsistency
- Do the changes in the same PR with the original issue
- Omit those trail whitespace and create an issue for that

**Comment by bnevis-i** [comment]
Miguel,

The project relies on tools like `gofmt` to check the formatting of submitted code (it is launched automatically as part of `make test`).

I just ran a local test and indeed `gofmt` will report on trailing whitespaces.

Can you provide a specific example of formatting consistency issues that `gofmt` won't catch?

**Reply by mglmc** [reply]
Ok, that makes sense, since I forgot to mention that the file full of trail whitespace is in the edgex-docs.

Here it is an example in https://github.com/edgexfoundry/edgex-docs/blob/02dccf74b39ee4f522cba6b237eddb3c0650652f/docs_src/examples/Ch-ExamplesAddingMQTTDevice.md

![Screenshot 2023-07-11 081536](https://github.com/edgexfoundry/community/assets/20265876/9ab47994-8118-4419-a13c-75162dc5d05d)

**Reply by lenny-goodell** [reply, marked as answer]
We don't worry about extra spaces in the Markdown files. These files are compiled by mkdocs to generate the GitHub pages for the documentation.

**Reply by mglmc** [reply]
Thank you for the answer
